File: 150-EvaluateReversePolishNotation/150-EvaluateReversePolishNotation.java
// Last updated: 9/14/2025, 12:54:06 PM
import java.util.Stack;

public class Solution {
    public int evalRPN(String[] tokens) {
        int a,b;
		Stack<Integer> S = new Stack<Integer>();
		for (String s : tokens) {
			if(s.equals("+")) {
				S.add(S.pop()+S.pop());
			}
			else if(s.equals("/")) {
				b = S.pop();
				a = S.pop();
				S.add(a / b);
			}
			else if(s.equals("*")) {
				S.add(S.pop() * S.pop());
			}
			else if(s.equals("-")) {
				b = S.pop();
				a = S.pop();
				S.add(a - b);
			}
			else {
				S.add(Integer.parseInt(s));
			}
		}	
		return S.pop();
	}
}

File: 150-EvaluateReversePolishNotation/150-EvaluateReversePolishNotation.py
# Last updated: 1/12/2026, 10:26:08 PM
1class Solution:
2    def evalRPN(self, tokens: List[str]) -> int:
3        stack = []
4        op_set= {'+','-','*','/'}
5        for ch in tokens:
6            if ch not in op_set:
7                stack.append(int(ch))
8            else:
9                right = stack.pop()
10                left = stack.pop()
11                if ch == '+':
12                    tmp = left + right
13                if ch == "-":
14                    tmp = left - right
15                if ch == "*":
16                    tmp = left*right
17                if ch == "/":
18                    if (left < 0 and right > 0) or (left > 0 and right < 0):
19                        tmp = abs(left)//abs(right) * (-1)
20                    else:
21                        tmp = left //right
22                stack.append(tmp)
23        return stack[-1]